src/View/Helper/PaginatorHelper.php

```diff
@@ -1152,11 +1152,27 @@ public function limitControl(array $limits = [], ?int $default = null, array $op
         $default ??= $this->paginated()->perPage();
         $scope = $this->param('scope');
         assert($scope === null || is_string($scope));
+
+        $url = null;
+        $currentPage = $this->paginated()->currentPage();
+
+        if ($currentPage > 1) {
+            $query = $this->_View->getRequest()->getQueryParams();
+
+            if ($scope) {
+                $query[$scope]['page'] = 1;
+            } else {
+                $query['page'] = 1;
+            }
+
+            $url = $this->_View->getRequest()->getPath();
+            $url .= '?' . http_build_query($query);
+        }
```

Comment:
Getting reset to page of one could be confusing for users though. If we have a total for the pagination result, could we use that to figure out where they are in the resultset. Once we know where they are, we can calculate a new first page so that they don't lose their spot. We can (and should) also do a bounds check.

Reply:
Yes calculating the new page number would be possible but that would need additional JS code to dynamically change the form's `action` before submission based on user selection.

Reply:
Ah true, what I suggested would be hard to do within the helper alone.